app/controllers/idv/in_person/usps_locations_controller.rb

```diff
@@ -27,8 +26,14 @@ def index
             response = proofer.request_pilot_facilities
           end
         rescue => err
-          Rails.logger.warn(err)
-          response = proofer.request_pilot_facilities
+          analytics.idv_in_person_locations_request_failure(
+            exception_class: err.class,
+            exception_message: err.message,
+            response_body_present: err.respond_to?(:response_body) && err.response_body.present?,
+            response_body: err.respond_to?(:response_body) && err.response_body,
+            response_status_code: err.respond_to?(:response_status) && err.response_status,
+          )
+          render json: {}, status: :internal_server_error and return
         end
 
         render json: response.to_json
```

Comment:
I think the `and return` could be easy to miss. Maybe we could move the happy path up into the `begin` to avoid the `return` altogether?

i.e.

```rb
begin
  # ...
  render json: response.to_json
rescue => err
  # ...
  render json: {}, status: :internal_server_error
end
```

Could also make it a method-level `rescue`:

```rb
def index
  # ...
  render json: response.to_json
rescue => err
  # ...
  render json: {}, status: :internal_server_error
end
```

Reply:
Good idea.  I'll refactor. 👍🏻